File: scripts/merger.R
################
# Merger : merge every feature in one unique dataset
################

### define feature names
features = c("Commenti","IscrittiConquistati","IscrittiPersi", "MiPiace","NonMiPiace","TempoDiVisualizzazione", "Visualizzazioni")

path = "./data/totals/"

for(feature in features){
  feature.path <- paste("./data/features/",feature, sep="")
  dirs <-  list.dirs(path = feature.path)
  dirs <- dirs[-1] # remove first redundant
  print(feature)
  
  feature.df = data.frame() # data frame for the feature
  
  for(dir in dirs){
    # extract file "Dati del grafico"
    files <- list.files(path=dir, full.names = TRUE)
    file.name <- files[1]
    df <- read.csv(file.name)
    # print(dim(df)[1])
    feature.df <- rbind(feature.df,df)
  }
  print(dim(feature.df)[1])
  print(summary(feature.df))
  write.csv(feature.df,paste(path,feature,"_totals.csv",sep=""),row.names = FALSE)
  print("---")
}

merge <- function(df1,df2){
  print(df1)
  print(df2)
  df <- data.frame()
  return(df)
}

print(typeof(dfs))

File: scripts/python/merger.py
import pandas as pd
import glob


features = ["Commenti", "IscrittiConquistati", "IscrittiPersi",
            "MiPiace", "NonMiPiace", "TempoDiVisualizzazione", "Visualizzazioni"]

columns_to_drop = ["Titolo video",
                   "Ora pubblicazione video"]

path_to_save = "../../data/merged/"

for feature in features:

    print("\n### FEATURE "+feature+"\n")

    # path for file in feature folder
    path = '..\\..\\data\\features\\'+feature+'\\'

    # find all matching file inside every feature folder
    files = [f for f in glob.glob(
        path + "**/Dati del grafico.csv", recursive=True)]

    # store all the dataframes for one feature
    feature_dfs = []
    total_df = pd.DataFrame()

    for f in files:
        # print("\tFile: "+f)

        # read csv
        df = pd.read_csv(f)

        # drop columns
        df = df.drop(columns=columns_to_drop)

        # append to list
        feature_dfs.append(df)

        if len(total_df) == 0:
            total_df = df
        else:
            total_df.append(df)
    total_df.to_csv(path_to_save+feature+"_total.csv", index=False)
